cmd/otk-make-partition-stages/main.go

```diff
@@ -0,0 +1,61 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/osbuild/images/pkg/disk"
+	"github.com/osbuild/images/pkg/osbuild"
+)
+
+// keep in sync with "otk-gen-partition-table"
+type Input struct {
+	Internal InputInternal `json:"internal"`
+}
```

Comment:
I'm thinking about the word `internal` and whether it's confusing to expose it.  Maybe if this becomes a convention with data structures it will be okay, but if it doesn't, I wonder if `filesystem.internal` creates some ambiguity (internal filesystem??).

This is a more general design decision I think.  I think we can keep using the idea of an `internal` data (sub)structure, but try out best to hide it from appearing in the yaml bits.  So, for example, our PT generator will still return
```json
{
  "const": {
    "internal": {
      <stuff>
    }
  }
}
```
but then instead of feeding just the `internal` bits back into the stage generator, it could be:
```yaml
stages:
 ...
 otk.external.make_partition_stages:
   filesystem: ${filesystem}
```

and then the stage generator loads and uses the `.internal` part, hiding the substructure which should be, for the otk user, an implementation detail.

Reply:
I like this! Given that we have https://github.com/achilleas-k/images/pull/8 lined up here too we could do it as a followup? I think it makes a ton of sene